tests/end2end/test_keras_converter.py

```diff
@@ -44,15 +47,41 @@ def test_custom_op(self):
         model.add(MaxPooling2D((2, 2), strides=(2, 2), data_format='channels_last'))
 
         model.compile(optimizer='sgd', loss='mse')
+        converted_model = onnxmltools.convert_keras(model, custom_conversion_functions={ScaledTanh: custom_activation})
+
         actual = model.predict(x)
         self.assertIsNotNone(actual)
 
-        converted_model = onnxmltools.convert_keras(model, custom_conversion_functions={ScaledTanh: custom_activation})
         self.assertIsNotNone(converted_model)
         # to check the model, you can print(str(converted_model))
-        dump_data_and_model(x.astype(np.float32), model, converted_model, basename="KerasCustomOp-Out0",
+        dump_data_and_model(x.astype(np.float32), model, converted_model, basename="Keras_custom_op-Out0",
                             context=dict(ScaledTanh=ScaledTanh))
 
+    def test_channel_last(self):
+        N, C, H, W = 2, 3, 5, 5
+        x = np.random.rand(N, H, W, C).astype(np.float32, copy=False)
+
+        model = keras.Sequential()
+        model.add(Conv2D(2, kernel_size=(1, 2), strides=(1, 1), padding='valid', input_shape=(H, W, C),
+                         data_format='channels_last'))  # , activation='softmax')
+        model.add(MaxPooling2D((2, 2), strides=(2, 2), data_format='channels_last'))
+
+        model.compile(optimizer='sgd', loss='mse')
+        converted_model = onnxmltools.convert_keras(model, channel_first_inputs=[model.inputs[0].name])
+
+        expected = model.predict(x)
+        self.assertIsNotNone(expected)
+        self.assertIsNotNone(converted_model)
+
+        try:
```

Comment:
Why not using dump_data_and_model?

Reply:
The inputs between keras and onnx model are different, which hasn't been supported by dump_data_and_model.

Reply:
Why not adding an extra parameter to support as transform (input_transform=lambda input: input.transpose( …)?